code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/identity/TokenResourceManager.java

```diff
@@ -131,6 +162,16 @@ public void setTokenCreators(List<TokenCreator> tokenCreators) {
         this.tokenCreators = tokenCreators;
     }
 
+    public void deletePreviousToken(long authenticatedAsAccountId, long tokenAccountId) {
+        List<AuthToken> authTokenList = authTokenDao.getTokenByAuthenticatedAsAccountId(authenticatedAsAccountId, tokenAccountId);
```

Comment:
@cjellick yes I can move this all to the DAO and do the delete as you said with a single query.